### PR TITLE
Add support for OpenSSL compatibility function SSL_CTX_get_min_proto_…

### PR DESCRIPTION
**Pull request by haydenroche5**
…version.

This is needed by socat-1.7.4.1.

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -16623,6 +16623,62 @@ int wolfSSL_CTX_set_max_proto_version(WOLFSSL_CTX* ctx, int ver)
     return sanityCheckProtoVersion(ctx);
 }
 
+WOLFSSL_API int wolfSSL_CTX_get_min_proto_version(WOLFSSL_CTX* ctx)
```
Comment (haydenroche5) [inline src/ssl.c:16626]: This essentially does the reverse of the mapping in wolfSSL_CTX_set_min_proto_version.

**Review by SparkiDev — APPROVED** [review]

File: src/ssl.c [hunk 2]
```diff
@@ -16623,6 +16623,90 @@ int wolfSSL_CTX_set_max_proto_version(WOLFSSL_CTX* ctx, int ver)
...
+            ret = TLS1_2_VERSION;
+            break;
+#endif
+#ifdef WOLFSSL_TLS13
+        case TLSv1_3_MINOR:
+            ret = TLS1_3_VERSION;
+            break;
+#endif
+        default:
+            ret = 0;
```
Comment (haydenroche5) [inline src/ssl.c:16657]: "The getter functions return the configured version or 0 for auto-configuration of lowest or highest protocol, respectively." (https://www.openssl.org/docs/man1.1.1/man3/SSL_get_min_proto_version.html)

File: src/ssl.c [hunk 3]
```diff
@@ -16623,6 +16623,91 @@ int wolfSSL_CTX_set_max_proto_version(WOLFSSL_CTX* ctx, int ver)
...
+{
+    int ret;
+
+    WOLFSSL_ENTER("wolfSSL_CTX_get_min_proto_version");
+
+    if (ctx == NULL) {
+        ret = get_default_min_proto_version(WOLFSSL_MIN_DOWNGRADE);
+    }
+    else {
+        switch (ctx->minDowngrade) {
```
Comment (SparkiDev) [inline src/ssl.c:16674]: Why not call gert_default_min_proto_version with ctx->minDowngrade and if 0, call with WOLFSSL_MIN_DOWNGRADE?

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/ssl.c (hunk 3 above)
Comment (haydenroche5) [reply src/ssl.c:16674]: Yeah, that's cleaner. I'll change.

Comment (SparkiDev) [reply src/ssl.c:16674]: Initialise ret to 0, check ctx not NULL and call function and then if ret is 0 call with define.

Comment (SparkiDev) [reply src/ssl.c:16674]: Still want to call with the define when ctx is NULL.

Comment (haydenroche5) [reply src/ssl.c:16674]: I think that just brings us back to what I had before, no?

Comment (haydenroche5) [reply src/ssl.c:16674]:
```
WOLFSSL_API int wolfSSL_CTX_get_min_proto_version(WOLFSSL_CTX* ctx)
{
    int ret = 0;

    WOLFSSL_ENTER("wolfSSL_CTX_get_min_proto_version");

    if (ctx == NULL) {
        ret = GetMinProtoVersion(WOLFSSL_MIN_DOWNGRADE);
    }
    else {
        ret = GetMinProtoVersion(ctx->minDowngrade);

        if (ret == 0) {
            ret = GetMinProtoVersion(WOLFSSL_MIN_DOWNGRADE);
        }
    }

    WOLFSSL_LEAVE("wolfSSL_CTX_get_min_proto_version", ret);

    return ret;
}
```

Comment (SparkiDev) [reply src/ssl.c:16674]: How about:
WOLFSSL_API int wolfSSL_CTX_get_min_proto_version(WOLFSSL_CTX* ctx)
{
    int ret = 0;

    WOLFSSL_ENTER("wolfSSL_CTX_get_min_proto_version");

    if (ctx != NULL) {
        ret = GetMinProtoVersion(ctx->minDowngrade);
    }
    if (ret == 0) {
        ret = GetMinProtoVersion(WOLFSSL_MIN_DOWNGRADE);
    }

    WOLFSSL_LEAVE("wolfSSL_CTX_get_min_proto_version", ret);

    return ret;
}

Comment (haydenroche5) [reply src/ssl.c:16674]: Looks good to me. Will re-push.

**Review by SparkiDev — APPROVED** [review]

**Review by toddouska — APPROVED** [review]

**Pull request merged**